Code/Framework/AzCore/AzCore/Jobs/Internal/JobTypeEraser.h

```diff
@@ -0,0 +1,232 @@
+    // static class function which has a stable address in memory. The Erased* methods return addresses to the
+    // indirect callers of the lambda copy/move assignment operators, call operator, and destructor.
+    //
+    // For lambdas that are trivially relocatable, both the returned move and copy assignment function pointers
+    // will be nullptr.
+    //
+    // Lambdas that are trivially destructible will result in a nullptr returned JobDestroy_t pointer.
+    //
+    // The class will check that the lambda is copy assignable or movable.
+    template<typename Lambda>
+    class JobTypeEraser final
+    {
+    public:
+        constexpr JobInvoke_t ErasedInvoker()
+        {
+            return reinterpret_cast<JobInvoke_t>(Invoker);
+        }
+
+        constexpr JobRelocate_t ErasedRelocator()
+        {
+            if constexpr (AZStd::is_trivially_move_constructible_v<Lambda>)
+            {
+                return nullptr;
+            }
+            else if constexpr (AZStd::is_move_constructible_v<Lambda>)
+            {
+                return reinterpret_cast<JobRelocate_t>(Mover);
+            }
+            else if constexpr (AZStd::is_copy_constructible_v<Lambda>)
+            {
+                return reinterpret_cast<JobRelocate_t>(Copyer);
+            }
+            else
+            {
+                static_assert(
+                    AZStd::is_move_constructible_v<Lambda> || AZStd::is_copy_constructible_v<Lambda>,
+                    "Job lambdas must be either move or copy constructible. Please verify that all captured data is move or copy "
+                    "constructible.");
+            }
+        }
+
+        constexpr JobDestroy_t ErasedDestroyer()
+        {
+            if constexpr (AZStd::is_trivially_destructible_v<Lambda>)
+            {
+                return nullptr;
+            }
+            else
+            {
+                return reinterpret_cast<JobDestroy_t>(Destroyer);
+            }
+        }
+
+    private:
+        constexpr static void Invoker(Lambda* lambda)
+        {
+            lambda->operator()();
+        }
+
+        constexpr static void Mover(Lambda* dst, Lambda* src)
+        {
+            new (dst) Lambda{ AZStd::move(*src) };
+        }
+
+        constexpr static void Copyer(Lambda* dst, Lambda* src)
+        {
+            new (dst) Lambda{ *src };
+        }
+
+        constexpr static void Destroyer(Lambda* lambda)
+        {
+            lambda->~Lambda();
+        }
+    };
+
+    // The TypeErasedJob encapsulates member function pointers to store in a homogeneously-typed container
+    // The function signature of all lambdas encoded in a TypeErasedJob is void(*)(). The lambdas can capture
+    // data, in which case the data is inlined in this structure if the payload is less than or equal to the
+    // buffer size. Otherwise, the data is heap allocated.
+    class alignas(alignof(max_align_t)) TypeErasedJob final
```

Comment:
I will say it's unlikely that this will be an issue and not warrant the need for overalignment for the majority of use cases. For one, 48 bytes of data is smaller than an actual AVX512 register, and for an AVX2 register, the cycle penalty of an unaligned load/store is not that severe. I think most people that do math ops will likely be capturing a &/this to a class/struct that contains the data itself.